ddtrace/internal/writer/writer.py

```diff
@@ -80,8 +84,7 @@ def _human_size(nbytes):
 
 class TraceWriter(metaclass=abc.ABCMeta):
     @abc.abstractmethod
-    def recreate(self):
-        # type: () -> TraceWriter
+    def recreate(self, appsec_enabled: Optional[bool] = None) -> "TraceWriter":
```

Comment:
I couldn't find a good way around this, since the value is coming from `tracer.configure` and not tied to a specific config option... we probably need to keep this until we can deprecate `appsec_enabled` from `tracer.configure()` :/